index.js

```diff
@@ -1,111 +1,153 @@
 const express = require('express');
-const server = express();
-const session = require('express-session');
-const bcrypt = require('bcryptjs');
 const cors = require('cors');
-const jwt = require('jsonwebtoken');
+const bcrypt = require('bcryptjs');
+const session = require('express-session')
 
+const db = require('./database/dbConfig.js');
+const jwt = require('jsonwebtoken')
+const secret = 'secret!';
 const PORT = 9876;
 
-const db = require('./database/dbHelpers');
-
-// middleware goes here
+const server = express();
 
 server.use(express.json());
 server.use(cors());
 server.use(session({
-    // configure express-session middleware
-      name: 'notsession', // default is connect.sid
-      secret: 'help will always be given at Hogwarts to those who ask for it.',
-      cookie: {
-        maxAge: 1 * 24 * 60 * 60 * 1000,
-      }, // 1 day in milliseconds
-      httpOnly: true, // don't let JS code access cookies. Browser extensions run JS code on your browser!
-      resave: false,
-      saveUninitialized: false,
-  }))
+  // configure express-session middleware
+    name: 'notsession', // default is connect.sid
+    secret: 'nobody tosses a dwarf!',
+    cookie: {
+      maxAge: 1 * 24 * 60 * 60 * 1000,
+    }, // 1 day in milliseconds
+    httpOnly: true, // don't let JS code access cookies. Browser extensions run JS code on your browser!
+    resave: false,
+    saveUninitialized: false,
+}))
 
-server.get('/', (req, res) => {
-    res.send(`Hello, world!!`);
-})
 
-function protected() {
+server.get('/', (req, res) => {
+  res.send(`It's Alive!`);
+});
+
+function protected(req, res, next){
+  // using jwts instead of sessions
+  // read the token from the authorization header 
+  const token = req.headers.authorization;
+
+  if (token) {
+    // verify the token
+    jwt.verify(token, secret, (err, decodedToken) => {
+      if (err) {
+        // token is invalid
+        res.status(401).json({message: 'Invalid token'})
+      } else {
+        // token is valid
+        next();
+      }
+    })
+  } else {
+    res.status(401).json({message: "no token provided"})
```

Comment:
Typically it's best to be consistent with single or double quotes because it does matter in some other languages, and it's a bit easier on the eyes and the mind if it's consistent. I typically stick to single quotes in JS.

Reply:
You could also avoid some of this nesting by doing your check a little differently:
```javascript
if (!token) {
  res.status(401).json({ message: 'no token provided' });
}

jwt.verify(token, secret, (err, decodedToken) => {
  if (err) {
    // token is invalid
    res.status(401).json({ message: 'Invalid token' })
  }
  // token is valid
  next();
})
```